src/NewProject/new_project_dialog.cpp

```diff
@@ -191,7 +191,66 @@ void newProjectDialog::ResetToProjectSettings() {
   m_addSrcForm->SetBasePath(m_projectManager->getProjectPath());
 }
 
+// This will check the project wizard for potential conflicts
+// If an errStr is passed in, the found issues will be appended
+// to it.
+// This function is intended to grow as more potential conflicts
+// are discovered.
+bool newProjectDialog::ValuesValid(QString *errStr /*nullptr*/) {
+  QString temp;
+  if (!errStr) {
+    errStr = &temp;
```

Comment:
Undefined behavior here

Reply:
I don't see fix of this problem. This is critical.

Reply:
Sorry, wrote a comment and hit resolve and forgot that it doesn't submit the comment like that. This is just to allow the user to ignore the errStr if they want. If they don't pass one, then we just point at a temp QString that we won't use (the errors will get written to avoid adding logic to every errStr +=, but the message itself won't be used besides for determining the bool status at the end). Let me know if i'm missing your point though.

Reply:
yes, you are right. But this looks like code smell. It is better to do like
if (errStr) *errStr = ...

Reply:
Wouldn't you to new and delete a QString then since errStr is a pointer?

Reply:
We don't need it. If pointer is not nullptr we do something if it is nullptr we do nothing. 
But there is better solution like: 
std::pair<bool, QString> ValuesValid() const;
and you don't need to care about pointers

Reply:
Forgot to hit comment again before resolve ;p
Sounds good, i had considered pair at first, but thought it might be overkill. I've refactored the code to use std::pair now.